addons/stock/models/stock_move.py

```diff
@@ -815,31 +809,84 @@ def onchange_date(self):
         if self.date_expected:
             self.date = self.date_expected
 
-    @api.onchange('move_line_nosuggest_ids')
-    def onchange_move_line_nosuggest_ids(self):
+    @api.onchange('move_line_ids', 'move_line_nosuggest_ids')
+    def onchange_move_line_ids(self):
         breaking_char = '\n'
-        move_lines_to_create = []
-        for move_line in self.move_line_nosuggest_ids:
+        if self.picking_type_id.show_reserved:
```

Comment:
this whole onchange that will mainly be used on receipt with the hidden move lines will now be executed on every changes (even the one without lot_name) on every move lines (even internal).. we 100% need to be able to shortcircuit this thing se the onchanges are not slower than they already are

encoding a simple lot_name where nothing has to be done goes from 0.6s to 1.6s on my local branch